docs/topics/class-based-views/advanced.txt

```diff
@@ -0,0 +1,694 @@
+
+    from django import models
+
+    class Book(models.Model):
+        name = models.CharField(max_length=50, unique=True)
+        author = models.ForeignKey(Author, related_name="books")
+
+Note that when using ``context_object_name`` property or an automatically added
+context variable with the name based on model name, we need to be mindful of
+the inheritance order: in this case we'll use the ``author`` context variable
+which is set up in ``SingleObjectMixin.get_context_data()`` so we have to put
+``SingleObjectMixin`` first.
+
+.. code-block:: html+django
+
+    <h2>Author {{ author }}</h2>
+
+    <ol>
+      {% for book in page_obj %}
+        <li>{{ book.name }}</li>
+      {% endfor %}
+    </ol>
+
+    <div class="pagination">
+        <span class="step-links">
+            {% if page_obj.has_previous %}
+                <a href="?page={{ page_obj.previous_page_number }}">previous</a>
+            {% endif %}
+
+            <span class="current">
+                Page {{ page_obj.number }} of {{ paginator.num_pages }}.
+            </span>
+
+            {% if page_obj.has_next %}
+                <a href="?page={{ page_obj.next_page_number }}">next</a>
+            {% endif %}
+        </span>
+    </div>
+
+Combined Detail and Form Views
+==============================
+
+As with the last example, let's look at what we need to reuse:
+``get_context_data()`` methods from both classes, and the ``post()`` method
+from ``FormView``.
+
+Unlike the last example, there is no clash between ``get()`` methods:
+``FormView``'s method does not need to do any special setup, therefore we can
+simply skip it by putting ``SingleObjectMixin`` first in the order of inheritance.
+
+As such, we don't need to override any methods at all! However, we may
+sometimes wish to override ``form_valid()`` to do some processing of the form
+(but would have to do the same in a stand-alone ``FormView``):
+
+.. snippet::
+    :filename: views.py
+
+    from django.views.generic import FormView
+    from django.shortcuts import redirect
+    from django import forms
+
+    class AuthorInterestForm(forms.Form):
+        message = forms.CharField()
+
+    class AuthorDetail2(SingleObjectMixin, FormView):
+        model = Author
+        form_class = AuthorInterestForm
+        template_name = "author2.html"
+
+        def form_valid(self, form):
+            # Here, we would have some logic based on user's form inputs
+            return redirect('author2', pk=self.get_object().pk)
+
+The template can use the ``author`` context variable and display the form:
+
+.. code-block:: html+django
+
+    <h2>Author {{ author.name }}</h2>
+
+    <form action="" method="post" accept-charset="utf-8">{% csrf_token %}
+        <table border="0" cellpadding="2" cellspacing="2">
+            {{ form.as_table }}
+        </table>
+    <p><input type="submit" value="Submit" /></p>
+    </form>
+
+Combined List View with a Form
+==============================
```

Comment:
Combining a list view with a completely unrelated form seems an unusual use case, not something to cover in as a common example.